lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2919,8 +2920,7 @@ static bool checkFunctionSignature(
 static IndexSubset *computeDifferentiationParameters(
     ArrayRef<ParsedAutoDiffParameter> parsedWrtParams,
     AbstractFunctionDecl *function, GenericEnvironment *derivativeGenEnv,
-    StringRef attrName, SourceLoc attrLoc
-) {
+    StringRef attrName, SourceLoc attrLoc, bool diagnoseErrors = true) {
```

Comment:
Rather than adding an invasive `diagnoseErrors` flag, can you try using `DiagnosticTransaction` for `diagnoseErrors = false` users instead?

`DiagnosticTransaction` collects all diagnostics and provides a way to cancel them:
```cpp
        DiagnosticTransaction transaction(Ctx.Diags);
        SWIFT_DEFER { transaction.abort(); };
```

See [#28717](https://github.com/apple/swift/pull/28717/files#diff-163f1b115cbb43d2e7bcc01e6c5eb2ceR3735) for an example.

Reply:
I'll try this.

Reply:
done